sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/MiscFunctionsSuite.scala

```diff
@@ -132,4 +132,69 @@ class MiscFunctionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       }
     }
   }
+
+  test("aesEncrypt") {
+    checkEvaluation(Base64(AesEncrypt(Literal("ABC".getBytes),
+      Literal("1234567890123456".getBytes))), "y6Ss+zCYObpCbgfWfyNWTw==")
+    checkEvaluation(Base64(AesEncrypt(Literal("".getBytes),
+      Literal("1234567890123456".getBytes))), "BQGHoM3lqYcsurCRq3PlUw==")
+
+    // input is null
+    checkEvaluation(AesEncrypt(Literal.create(null, BinaryType),
+      Literal("1234567890123456".getBytes)), null)
+    // key is null
+    checkEvaluation(Base64(AesEncrypt(Literal("ABC".getBytes),
+      Literal.create(null, BinaryType))), null)
+    // both are null
+    checkEvaluation(Base64(AesEncrypt(Literal.create(null, BinaryType),
+      Literal.create(null, BinaryType))), null)
+
+    // key length (80 bits) is not one of the permitted values (128, 192 or 256 bits)
+    intercept[java.security.InvalidKeyException] {
+      evaluate(AesEncrypt(Literal("ABC".getBytes), Literal("1234567890".getBytes)))
+    }
+  }
+
+  test("aesDecrypt") {
+    checkEvaluation(AesDecrypt(UnBase64(Literal("y6Ss+zCYObpCbgfWfyNWTw==")),
+      Literal("1234567890123456".getBytes)), "ABC")
+    checkEvaluation(AesDecrypt(UnBase64(Literal("BQGHoM3lqYcsurCRq3PlUw==")),
+      Literal("1234567890123456".getBytes)), "")
+
+    // input is null
+    checkEvaluation(AesDecrypt(UnBase64(Literal.create(null, StringType)),
+      Literal("1234567890123456".getBytes)), null)
+    // key is null
+    checkEvaluation(AesDecrypt(UnBase64(Literal("y6Ss+zCYObpCbgfWfyNWTw==")),
+      Literal.create(null, BinaryType)), null)
+    // both are null
+    checkEvaluation(AesDecrypt(UnBase64(Literal.create(null, StringType)),
+      Literal.create(null, BinaryType)), null)
+
+    // key length (80 bits) is not one of the permitted values (128, 192 or 256 bits)
+    intercept[java.security.InvalidKeyException] {
+      evaluate(AesDecrypt(UnBase64(Literal("y6Ss+zCYObpCbgfWfyNWTw==")),
```

Comment:
@cloud-fan I added codegen version tests with creating unsafe projection. But I'm not very sure whether I am doing right or not. Could you take a look again?
Meanwhile, I took look at code `ExpressionEvalHelper.scala` [`checkEvalutionWithUnsafeProjection()`](https://github.com/apache/spark/blob/master/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvalHelper.scala#L123-L149). Is that doing the same thing as you said to test codegen version?